boards/arm/arty/CMakeLists.txt

```
@@ -0,0 +1,23 @@
+# SPDX-License-Identifier: Apache-2.0
+
+if((CONFIG_BOARD_ARTY_A7_ARM_DESIGNSTART_M1) AND (CONFIG_BUILD_OUTPUT_BIN))
```

should this have been ?:
```suggestion
if((CONFIG_BOARD_ARTY_A7_ARM_DESIGNSTART_M1) AND
   (CONFIG_BUILD_OUTPUT_BIN) AND
   (CONFIG_BUILD_OUTPUT_HEX)
)
```

after all, what is the point in looking for `bin2hex` if `CONFIG_BUILD_OUTPUT_HEX=n`.

---

No, it needs to depend on `CONFIG_BUILD_OUTPUT_BIN` since the .bin file is used as input for generating the .mem file. While the .mem file is a hex dump of the memory, it is in a different format (Verilog memory hex dump) than the hex file (intel hex format) generated when enabling `CONFIG_BUILD_OUTPUT_HEX`.

---

Thanks for explaining.

Of course the bin file is still a dependency, which is why I kept it in my code suggestion ^^^.

I'm also aware that the verilog hex format is different from the intel hex format, but I assume the intel hex format is of no interest on this target, therefore I simply wondered if `CONFIG_BUILD_OUTPUT_HEX` should be used to indicate verilog hex format on this target.

Or maybe we should provide an additional setting here:
https://github.com/zephyrproject-rtos/zephyr/blob/master/Kconfig.zephyr#L295-L319
after all, we do support `intel hex`, `bin`, `s19`, `exe` so maybe a verilog format would make sense for more targets in future.

But I let you decide on this.

The updated messages looks good.

---

I plan on submitting proper, tree-wide support for Verilog memory hex dump format outputs once the binutils Verilog bfd bug (https://sourceware.org/bugzilla/show_bug.cgi?id=25202) is resolved. I have a branch ready for it here: https://github.com/henrikbrixandersen/zephyr/tree/verilog-memory-hex-dump

The board-specific solution here is just a temporary work-around for now. I would rather not interfere with `CONFIG_BUILD_OUTPUT_HEX` since I believe that might confuse users even more. The requirement on `CONFIG_BUILD_OUTPUT_BIN` for the Verilog .mem file generation has been added to the board documentation.

---

sounds great.